CONTRIBUTING.md

```diff
@@ -89,14 +89,6 @@ react-native init --version ${RN_VERSION}
 npm config set registry https://registry.npmjs.org/
 ```
 
-## Running `start` command
-
-In order for symlinks to work correctly when running `start` locally, set REACT_NATIVE_APP_ROOT as the root folder of your cli project:
-
-```
-REACT_NATIVE_APP_ROOT=path/to/cli node path/to/cli/packages/cli/build/index.js start
```

Comment:
I wonder how you test the `start` or any command that calls Metro now? This was a hack for Metro to "see" files that are symlinked alongside the cwd you're in. In Metro pre <0.57 you could also see these paths displayed in the UI (normally it's your CWD, but with this env passed it's CWD + this specified dir and its subdirs)

Reply:
Hm, isn't this what `projectRoots` and `watchFolders` stands for? 

Also, following up with this example:
```
REACT_NATIVE_APP_ROOT=path/to/cli node path/to/cli/packages/cli/build/index.js start
```

why Metro would care about symlinks that are in `path/to/cli`, if it's working in `cwd` where your React Native application business code is? Metro is not calling into `path/to/cli`.

Reply:
Also, the flag name is `REACT_NATIVE_APP_ROOT` which suggests you're setting application root. In the example, it's the CLI root, not the application root. It's also confusing since you can set a project root in a different way.

Reply:
FWIW, I have never used this way to debug `react-native start` command locally and it worked flawlessly.

However, as I stated in my comment [here](https://github.com/react-native-community/cli/pull/935#issuecomment-580274130), I am trying to understand the original intent and use case for that code. Looks like we brought it over from React Native.

Was it prior to `--projectRoots`? Or `--watchFolders`? We scan for symlinks when this environmental variable is present and that functionality seems useful. However, in 99.9% cases we do not do it.

Reply:
There you go https://github.com/facebook/react-native/commit/a1e105eea51ec7b8962bd43eec8be8aa15d94c5b

Reply:
> why Metro would care about symlinks that are in path/to/cli, if it's working in cwd where your React Native application business code is? Metro is not calling into path/to/cli.

It does for me locally, when I link local build of CLI into a RN project with `yarn link <cli_packages>` – Metro tries to resolve some files, like some babel plugin I don't remember now, from my local CLI dir.

So with this PR, you're removing that possibility, and essentially `resolveSymlinksForRoots` and `findSymlinkedModules` become dead code (should be removed if we want to go that path). 

I just don't want to end up in a position where we can't test local changes to CLI commands invoking Metro on a real project.

Reply:
> It does for me locally, when I link local build of CLI into a RN project with yarn link <cli_packages> – Metro tries to resolve some files, like some babel plugin I don't remember now, from my local CLI dir.

Can you try just setting `--watchFolders` flag when running `start`? If you can reproduce it of course. 

My guess is this: it's project roots / watch folders that were making this work, not `findSymlinkedModules`. I don't feel the urge of doing anything with this code, but I just want to understand the intents behind it, because it's really confusing.

> So with this PR, you're removing that possibility, and essentially resolveSymlinksForRoots and findSymlinkedModules become dead code (should be removed if we want to go that path).

Keep in mind, as per my different thread, that we used to call this no matter what. I have started off a different thread to understand the purpose of that file.

Reply:
Will try to find time to test it as well. I just remember I spent quite a bit of time to solve testing local changes to CLI `start` in a test project. I would like you to actually test it as well and feel the experience of changing anything in Metro-related code of CLI and `run-ios`/`run-android`/`start` a sample project seeing if Metro version local to this project picks it up or crashes because it can't find some module